### ISSUE TITLE
[Bug] Metadata DB removes items from Places DB via referrer_place_id

### ISSUE THREAD

**Issue by mavduevskiy** [mozilla/application-services#4808]
Relates to [https://github.com/mozilla-mobile/fenix/issues/23517](..fenix/issue/23517). 

Background: deleting items from history used to be not very consistent, and to align the behavior with the desktop, we are actually removing the url from the places DB. Removing an item from the `moz_places` also removes related items from the `moz_places_metadata`, but items from `moz_places_metadata` also removed related to them items from `moz_places` by referrer_place_id. 

So, basically, the schema should be changed and data migrated.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/SYNC-3038)

**Comment by mavduevskiy** [comment]
Working on that, can not assign it to myself.

**Comment by mhammond** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1866511

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
